appserver/core/javaee-kernel/src/main/java/org/glassfish/kernel/javaee/WebContainerStarter.java

```diff
@@ -162,23 +162,17 @@
      * that can be handled only by the web container, and if so, starts
      * the web container
      */ 
+    @Override
     public void postConstruct() {
         domainProvider.get();
         Config serverConfig = serverConfigProvider.get();
 
-        boolean isStartNeeded = false;
-        if (serverConfig != null) {
-            if (isStartNeeded(serverConfig.getHttpService())) {
-                isStartNeeded = true;
-            }
-            if (!isStartNeeded && isStartNeeded(serverConfig.getNetworkConfig())) {
-                isStartNeeded = true;
-            }
-        }
+        boolean isStartNeeded = serverConfig != null
+                && (isStartNeeded(serverConfig.getHttpService()) || isStartNeeded(serverConfig.getNetworkConfig()));
 
         if (isStartNeeded) {
             startWebContainer();
-        } else {
+        } else if (serverConfig != null){
```

Comment:
It seems logical that if the `serverConfig` is null (as suggested to be possible by `isStartNeeded`) we simply cannot do the initialisation. Behaviour change: NPE becomes NOOP

Reply:
But I cannot imagine a situation where serverConfig is not provided. I think this is worth an archeological expedition on why server config could be optional.

Reply:
Can't give much insight into when a `serverConfig` would be null. This is purely based on preexisting code suggesting that it could be.